tests/test_containers/test_kubectl.py

```
@@ -121,3 +122,43 @@ def test_start_shell_sso(mock_refresh, kubectl_container):
     # make sure we are pointing to the right AWS credentials: /tmp/ folder for SSO
     assert container_args["environment"]["AWS_CONFIG_FILE"] == "/home/leverage/tmp/test/config"
     assert container_args["environment"]["AWS_SHARED_CREDENTIALS_FILE"] == "/home/leverage/tmp/test/credentials"
+
+
+def test_scan_clusters(kubectl_container: KubeCtlContainer):
+    """
+    Test that we can find valid metadata.yaml presents in the down the path of the filesystem tree where we are staying.
+    """
+    # mock and call
+    with mock.patch("os.walk") as mock_walk:
+        with patch("builtins.open"):
```

Ruff (0.7.0)

Use a single `with` statement with multiple contexts instead of nested `with` statements

Combine `with` statements

(SIM117)